### PR TITLE
Added 'deattach_from' to 'ActiveSupport::Subscriber' to deattach a su…

### PR DESCRIPTION
**Pull request by sushantmittal**
Not ready. Checking why test cases of https://github.com/rails/rails/pull/35691 failing in travis ci. They passed in my local environment.

### CODE REVIEW

File: activesupport/lib/active_support/subscriber.rb [hunk 1]
```diff
@@ -58,23 +78,49 @@ def subscribers
...
+
+          notifier.unsubscribe(subscriber.patterns[pattern])
+          subscriber.patterns.delete(pattern)
+        end
+
+        def find_attached_subscriber(subscriber)
+          subscribers.find { |attached_subscriber| attached_subscriber.instance_of?(subscriber.class) }
+        end
+
+        def is_invalid_event?(event)
```
Comment (simi) [inline activesupport/lib/active_support/subscriber.rb]: Isn't `invalid_event?(event)` enough? IMHO it is not common to use `is_` (same as `get_`) as method prefixes in Ruby.

File: activesupport/test/subscriber_test.rb [hunk 2]
```diff
@@ -53,4 +68,26 @@ def test_does_not_attach_private_methods
 
     assert_equal [], TestSubscriber.events
   end
+
+  def test_deattaches_subscribers
+    TestSubscriber.deattach_from :doodle
+
+    sleep(60 * 2)
```
Comment (simi) [inline activesupport/test/subscriber_test.rb]: Isn't there any way how to remove this sleep? Can't be `Concurrent::CountDownLatch` used for example?

Comment (sushantmittal) [reply activesupport/test/subscriber_test.rb]: I am still debugging it. Main pull request is https://github.com/rails/rails/pull/35691

Test cases are failing in travis ci but passed in my local.
It just passed after adding `sleep`. Yeah I will try `Concurrent::CountDownLatch` as well. Thanks.

File: activesupport/lib/active_support/subscriber.rb (hunk 1 above)
Comment (sushantmittal) [reply activesupport/lib/active_support/subscriber.rb]: ok I will update it. Thanks.

File: activesupport/test/subscriber_test.rb (hunk 2 above)
Comment (simi) [reply activesupport/test/subscriber_test.rb]: Ahh, sorry. I'll move my comments to another PR.

**Comment by rafaelfranca** [comment]
There is no need to open a new PR for the same change.

**Pull request closed without merging**